packages/namekit-react/src/components/Input.tsx

```diff
@@ -60,26 +62,34 @@ export const Input: React.FC<InputProps> & { Slot: React.FC<SlotProps> } = ({
     variantClasses[variant],
     sizeClasses[inputSize],
     className,
+    {
+      "nk-border-red-300": !!error,
+    },
   ]);
 
   const inputClasses = cc([
-    "nk-ring-0 focus:nk-outline-none nk-w-full h-full placeholder:nk-text-gray-500 nk-bg-transparent",
+    "nk-ring-0 focus:nk-outline-none nk-w-full h-full placeholder:nk-text-gray-500 nk-bg-transparent nk-border-red-300",
     hasSlot && slotPosition === "left" ? "nk-pl-2" : "nk-pr-2",
   ]);
 
   return (
-    <div className={combinedClassName}>
-      {hasSlot && slotPosition === "left" && (
-        <label htmlFor={inputId} className={slotClassName}>
-          {children}
-        </label>
-      )}
-      <input id={inputId} className={inputClasses} {...props} />
-      {hasSlot && slotPosition === "right" && (
-        <label htmlFor={inputId} className={slotClassName}>
-          {children}
-        </label>
-      )}
+    <div className="nk-gap-1">
+      <div className={combinedClassName}>
+        {hasSlot && slotPosition === "left" && (
+          <label htmlFor={inputId} className={slotClassName}>
+            {children}
+          </label>
+        )}
+        <input id={inputId} className={inputClasses} {...props} />
+        {hasSlot && slotPosition === "right" && (
+          <label htmlFor={inputId} className={slotClassName}>
+            {children}
+          </label>
+        )}
+      </div>
+      <span className="nk-mt-2 nk-text-sm nk-font-normal nk-text-red-600">
```

Comment:
Maybe not relevant, since an empty span is fine as I understand it? But perhaps we first check if `error` is defined before we emit this span and its related nested logic?

Reply:
@lightwalker-eth Sounds good